crates/client/mempool/src/inner.rs

```diff
@@ -174,7 +176,7 @@ impl NonceChain {
             self.front_nonce = mempool_tx_nonce;
             #[cfg(debug_assertions)]
             {
-                self.front_tx_hash = mempool_tx_hash;
+                self.front_tx_hash = mempool_tx.tx_hash();
```

Comment:
Why not `mempool_tx_hash`? This is incurring clones even when this feature is disabled.